### PR TITLE
[epicli upgrade] Preserve existing Xmx for Elasticsearch

### PR DESCRIPTION
**Pull request by to-bar**
- Preserve existing Xmx for upgrade
- Perform Elasticsearch upgrade when needed
- Update how-to doc
- Fix Kibana upgrade when legacy version is present

### CODE REVIEW

**Review by rafzei — APPROVED** [review]
Looks Fantastic To Me

**Review by seriva — APPROVED** [review]

**Review by rafzei — DISMISSED** [review]

File: core/src/epicli/data/common/ansible/playbooks/roles/opendistro_for_elasticsearch/templates/jvm.options.j2 [hunk 1]
```diff
@@ -22,8 +22,8 @@
 # Xmx represents the maximum size of total heap space
 
 {# Xms = Xmx by purpose, must be equal #}
--Xms{{ specification.jvm_options.Xmx }}
--Xmx{{ specification.jvm_options.Xmx }}
+-Xms{{ xmx }}
+-Xmx{{ xmx }}
```
[the remark this answers is not in the document]
Comment (to-bar) [reply core/src/epicli/data/common/ansible/playbooks/roles/opendistro_for_elasticsearch/templates/jvm.options.j2:26]: The answer is in the comments above 😉

**Review by rafzei — APPROVED** [review]

**Review by przemyslavic — APPROVED** [review]

File: core/src/epicli/data/common/ansible/playbooks/roles/opendistro_for_elasticsearch/templates/jvm.options.j2 (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkyc) [reply core/src/epicli/data/common/ansible/playbooks/roles/opendistro_for_elasticsearch/templates/jvm.options.j2:26]: Answering my own question with this quote: 

> Setting -Xms and -Xmx to the same value increases predictability by removing the most important sizing decision from the virtual machine. However, the virtual machine is then unable to compensate if you make a poor choice.

From [here](https://docs.oracle.com/javase/8/docs/technotes/guides/vm/gctuning/sizing.html)

I just hope we do right decisions when assigning values, though.

[the remark this answers is not in the document]
Comment (to-bar) [reply core/src/epicli/data/common/ansible/playbooks/roles/opendistro_for_elasticsearch/templates/jvm.options.j2:26]: From [Elasticsearch docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/heap-size.html):

> Elasticsearch will assign the entire heap specified in jvm.options via the Xms (minimum heap size) and Xmx (maximum heap size) settings. You should set these two settings to be equal to each other.

**Pull request merged**